src/management-system-v2/lib/data/file-manager-facade.ts

```diff
@@ -0,0 +1,435 @@
+  mimeType: string,
+  fileName: string,
+  fileContent?: Buffer | Uint8Array | Blob,
+) {
+  if (!isContentTypeAllowed(mimeType)) {
+    throw new Error(`Content type '${mimeType}' is not allowed`);
+  }
+
+  switch (entityType) {
+    case EntityType.PROCESS:
+      return saveProcessArtifact(entityId, fileName, mimeType, fileContent);
+    case EntityType.ORGANIZATION:
+      return saveOrganizationLogo(entityId, fileName, mimeType, fileContent);
+    // Extend for other entity types if needed
+    default:
+      throw new Error(`Unsupported entity type: ${entityType}`);
+  }
+}
+
+// Retrieve a file associated with an entity
+export async function retrieveEntityFile(
+  entityType: EntityType,
+  entityId: string,
+  fileName?: string,
+) {
+  switch (entityType) {
+    case EntityType.PROCESS:
+      if (!fileName) throw new Error('File name is required for process artifacts');
+      return retrieveProcessArtifact(entityId, fileName);
+    case EntityType.ORGANIZATION:
+      return getOrganizationLogo(entityId);
+    // Extend for other entity types if needed
+    default:
+      throw new Error(`Unsupported entity type: ${entityType}`);
+  }
+}
+
+// Delete a file associated with an entity
+export async function deleteEntityFile(
+  entityType: EntityType,
+  entityId: string,
+  fileName?: string,
+): Promise<boolean> {
+  switch (entityType) {
+    case EntityType.PROCESS:
+      if (!fileName) throw new Error('File name is required for process artifacts');
+      return deleteProcessArtifact(fileName);
+    case EntityType.ORGANIZATION:
+      return deleteOrganizationLogo(entityId);
+    // Extend for other entity types if needed
+    default:
+      throw new Error(`Unsupported entity type: ${entityType}`);
+  }
+}
+
+interface SaveProcessArtifactOptions {
+  generateNewFileName?: boolean;
+  useDefaultArtifactsTable?: boolean;
+  versionCreatedOn?: string;
+  replaceFileContentOnly?: boolean;
+}
+
+// Functionality for handling process artifact files
+export async function saveProcessArtifact(
+  processId: string,
+  fileName: string,
+  mimeType: string,
+  fileContent?: Buffer | Uint8Array | Blob,
+  options: SaveProcessArtifactOptions = {},
+) {
+  const {
+    generateNewFileName = true,
+    useDefaultArtifactsTable = true,
+    versionCreatedOn,
+    replaceFileContentOnly = false,
+  } = options;
+
+  const newFileName = generateNewFileName ? getNewFileName(fileName) : fileName;
+  const artifactType = getFileCategory(fileName, mimeType);
+  const filePath = generateProcessFilePath(newFileName, processId, mimeType, versionCreatedOn);
+
+  const usePresignedUrl = ['images', 'others'].includes(artifactType);
+
+  const { presignedUrl, status } = await saveFile(filePath, mimeType, fileContent, usePresignedUrl);
+
+  if (replaceFileContentOnly) {
+    //skip db update
+    return { presignedUrl, status };
+  }
+
+  if (!status) {
+    await deleteFile(filePath);
+    throw new Error('Failed to save file');
+  }
```

Comment:
it would rewrite a file, if it already exists. All files are unique due to UUID, if the replace flag is true it would use the passed filename to overwrite the file.